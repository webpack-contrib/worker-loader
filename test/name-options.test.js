import {
  compile,
  getCompiler,
  getErrors,
  getModuleSource,
  getResultFromBrowser,
  getWarnings,
} from './helpers';

describe('"name" option', () => {
  it('should work', async () => {
    const compiler = getCompiler('./basic/entry.js', {
      name: 'my-custom-name.js',
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./basic/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
