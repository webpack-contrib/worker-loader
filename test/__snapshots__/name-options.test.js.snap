// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"name" option should work: errors 1`] = `Array []`;

exports[`"name" option should work: module 1`] = `
"module.exports = function() {
  return new Worker(__webpack_public_path__ + \\"my-custom-name.js\\");
};"
`;

exports[`"name" option should work: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"name" option should work: warnings 1`] = `Array []`;
